### PR TITLE
Implement mongoose model encounter.model.ts

### PR DESCRIPTION
**Pull request by sliu929**
Added a mongoose model to represent an encounter in MongoDB database.

### CODE REVIEW

File: backend/src/models/encounter.model.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { Schema, model } from 'mongoose';
+
+interface EncounterModel {
+  date: Date,
+  location: string,
+  description: string,
+  persons: []
```
Comment (sdiv877) [inline backend/src/models/encounter.model.ts:7]: Might be good to use the `Schema.Types.ObjectId` type for the persons array in the interface

**Review by sdiv877 — CHANGES_REQUESTED** [review]

File: backend/src/models/encounter.model.ts (hunk 1 above)
Comment (sdiv877) [inline backend/src/models/encounter.model.ts:7]: Might be good to add the `Schema.Types.ObjectId` type to the `persons` array in the interface as well

**Review by sdiv877 — CHANGES_REQUESTED** [review]

File: backend/src/models/encounter.model.ts (hunk 1 above)
Comment (sliu929) [reply backend/src/models/encounter.model.ts]: will do

**Review by sdiv877 — APPROVED** [review]
LGTM 👍

File: backend/src/models/encounter.model.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  location: string,
+  description: string,
+  persons: [Schema.Types.ObjectId]
+}
+
+const schema = new Schema<EncounterModel>({
+  date: { type: Date, default: Date.now },
+  location: { type: String, required: false },
+  description: { type: String, required: true },
+  persons: { type: [Schema.Types.ObjectId], required: true }
```
Comment (butterplup) [inline backend/src/models/encounter.model.ts:14]: does making the `persons` of type `Schema.Types.ObjectId` mean it's storing the whole Person object or is it just a reference to a person object's ID?

Comment (sliu929) [reply backend/src/models/encounter.model.ts:14]: Just the reference ID.

Comment (butterplup) [reply backend/src/models/encounter.model.ts:14]: Sweet, that makes sense.

**Review by butterplup — APPROVED** [review]
Looks good

**Pull request merged**